sklearn/kernel_approximation.py

```diff
@@ -713,22 +736,35 @@ def transform(self, X):
             Whether the return value is an array or sparse matrix depends on
             the type of the input X.
         """
-        msg = (
-            "%(name)s is not fitted. Call fit to set the parameters before"
-            " calling transform"
-        )
-        check_is_fitted(self, msg=msg)
-
         X = self._validate_data(X, accept_sparse="csr", reset=False)
         check_non_negative(X, "X in AdditiveChi2Sampler.transform")
         sparse = sp.issparse(X)
 
+        if self.sample_interval is None:
```

Comment:
Here, we should instead detect if the user already called fit or not. If not then, we do the step below. So I would expect something like:

```python
if hasattr(self,` "_sample_interval"):
    # TODO: remove this branch in 1.5
    sample_interval = self._sample_interval
else:
    ... # The normal code duplicated from fit
    if self.sample_interval is None:
        ...

```